editor/editor_node.cpp

```diff
@@ -301,14 +301,23 @@ void EditorNode::disambiguate_filenames(const Vector<String> p_full_paths, Vecto
 				// and the scene name first to remove extensions so that this
 				// comparison actually works.
 				String path = p_full_paths[E->get()];
+
+				// Get rid of file extensions and res:// prefixes.
+				if (scene_name.rfind(".") >= 0) {
+					scene_name = scene_name.substr(0, scene_name.rfind("."));
+				}
 				if (path.begins_with("res://")) {
 					path = path.substr(6);
 				}
 				if (path.rfind(".") >= 0) {
 					path = path.substr(0, path.rfind("."));
 				}
-				if (scene_name.rfind(".") >= 0) {
-					scene_name = scene_name.substr(0, scene_name.rfind("."));
+
+				// Normalize trailing slashes when normalizing directory names.
+				if (scene_name.rfind("/") == scene_name.length() - 1 && path.rfind("/") != path.length() - 1) {
+					path = path + "/";
+				} else if (scene_name.rfind("/") != scene_name.length() - 1 && path.rfind("/") == path.length() - 1) {
+					scene_name = scene_name + "/";
 				}
```

Comment:
Simpler?
```suggestion
				// Normalize trailing slashes when normalizing directory names.
				scene_name = scene_name.trim_suffix("/");
				path = path.trim_suffix("/");
```

Reply:
Could actually be applied to both places I think? Will test.